pkg/kubelet/kubelet.go

```diff
@@ -1943,7 +1943,7 @@ func (kl *Kubelet) syncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 	// and BEFORE the pod status is changed on the API server
 	// to avoid race conditions with the resource deallocation code in kubernetes core.
 	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
-		if err := kl.containerManager.UnprepareResources(pod); err != nil {
+		if err := kl.containerManager.UnprepareDynamicResources(pod); err != nil {
 			return err
 		}
 	}
```

Comment:
Looks like this is the reason ... Let's wrap this in an `UnprepareDynamicResources` resources call to remain symmetrical in the API.

Reply:
thanks for pointing this out to me. Done.